### PR TITLE
Fix trailing whitespaces

### PR DESCRIPTION
**Pull request by Youssef1313**
Related to #40887 and #40884

### CODE REVIEW

File: docs/design/coreclr/jit/ryujit-tutorial.md [hunk 1]
```diff
@@ -606,7 +606,7 @@ public static int PopCount(ulong bitVectorArg)
 
 #### Notes
 The sample I'm going to walk through implements support for pop count (counting the number of '1' bits in a 64-bit value).
- 
+
 We're going to start by assuming that we have a method with a known signature that implements PopCount.
 Here's the implementation we're going to use. It simply takes the input value, and keeps anding with one, and then shifting right.
```
Comment (Youssef1313) [inline docs/design/coreclr/jit/ryujit-tutorial.md:611]: This have a visual change, which I think is okay.

**before removing the space**: The two sentences appeared in the same line.
**after removing the space**: The two sentences appeared in separate line. I think this is the desired behavior.

File: docs/design/coreclr/profiling/davbr-blog-archive/Attach.md [hunk 2]
```diff
@@ -96,12 +96,12 @@ So here's the catch.  What if a V4 app starts up in background GC mode _without_
...
+|
 
-\<configuration\>   
-    \<runtime\>   
-        \<gcConcurrent enabled="false"/\>   
-    \</runtime\>   
+\<configuration\>
+    \<runtime\>
+        \<gcConcurrent enabled="false"/\>
+    \</runtime\>
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Attach.md:104]: This have **undesired** visual change.

@stephentoub, this is an example of a problem that trailing whitespace option in editorconfig can cause in the absence of a markdownlint.

File: docs/design/coreclr/profiling/davbr-blog-archive/DoStackSnapshot - Callback CONTEXT Registers.md [hunk 3]
```diff
@@ -5,9 +5,9 @@ In my initial [post](DoStackSnapshot - Exception Filters.md) about DoStackSnapsh
 
 The quick answer is that **nonvolatile (i.e., preserved), integer registers** should be valid.  You don't really need many registers to walk the stack anyway.  Obviously, you want a good stack pointer and instruction pointer.  And hey, a frame pointer is handy when you come across an EBP-based frame in x86 (RBP on x64).  These are all included in the set, of course.  Specifically by architecture, you can trust these fields in your context:
 
-x86: Edi, Esi, Ebx, Ebp, Esp, Eip  
-x64: Rdi, Rsi, Rbx, Rbp, Rsp, Rip, R12:R15  
+x86: Edi, Esi, Ebx, Ebp, Esp, Eip
+x64: Rdi, Rsi, Rbx, Rbp, Rsp, Rip, R12:R15
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/DoStackSnapshot - Callback CONTEXT Registers.md:9]: This is again an **undesired** visible change. The two trailing spaces were treated as a `<br>`. After removing them, these appear in the same line.

File: docs/design/coreclr/profiling/davbr-blog-archive/DoStackSnapshot - Exception Filters.md [hunk 4]
```diff
@@ -7,38 +7,38 @@ For those of you diehard C# fans, you might be unaware of the existence of excep
...
+```
+
+The filters are the things that come after "When". We all know that, when an exception is thrown, its type must match the type specified in a Catch clause in order for that Catch clause to be executed. "When" is a way to further restrict whether a Catch clause will be executed. Now, not only must the exception's type match, but also the When clause must evaluate to True for that Catch clause to be chosen. In the example above, when we run, we'll skip the first Catch clause (because its filter returned False), and execute the second, thus showing a message box with "Positive" in it.
+
+The thing you need to realize about DoStackSnapshot's behavior (indeed, CLR in general) is that the execution of a When clause is really a separate function call. In the above example, imagine we take a stack snapshot while inside Positive(). Our managed-only stack trace, as reported by DoStackSnapshot, would then look like this (stack grows up):
+
+Positive
+Main
+Thrower
+Main
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/DoStackSnapshot - Exception Filters.md:41]: This is again **undesired**.

**Comment by Youssef1313** [comment]
I haven't reviewed all the changes. Just picked some files and pointed to examples of undesired changes, and one example of a possibly **desired** change.

cc: @jkotas @stephentoub What's the action you recommend to do?

File: docs/design/coreclr/profiling/davbr-blog-archive/Attach.md (hunk 2 above)
Comment (jkotas) [reply docs/design/coreclr/profiling/davbr-blog-archive/Attach.md:104]: This is poorly formatted both before and after this change. This should use ```xml instead.

**Comment by jkotas** [comment]
Could you please fixup the deltas where you can easily tell that they are not an improvement?

These types of cleanup tend to come with some damage and it is ok. It is hard to be perfect, but we can at least try to reduce the damage.

**Comment by jkotas** [comment]
Thank you for cleaning this up!

File: docs/design/coreclr/profiling/davbr-blog-archive/Debugging - Activation.md [hunk 5]
```diff
@@ -15,7 +15,7 @@ Environment variables --\> Registry --\> Profiler DLL on File system.
 
 The first link in this chain is to check the environment variables inside the process that should be profiled.  If you're running the process from a command-prompt, you can just try a "set co" from the command prompt:
 
-| 
+|
 ```
 **C:\>** set co
  (blah blah, other vars beginning with "co")
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Debugging - Activation.md:21]: This is unrelated to the whitespace changes, but are the asterisk inside the code fence desired? they won't make it bold.

Also, the vertical bar in the line before the code fence, it seems extra.

File: docs/design/coreclr/profiling/davbr-blog-archive/Debugging - SOS and IDs.md [hunk 6]
```diff
@@ -9,7 +9,7 @@ SOS.DLL is a debugger extension DLL that ships with the CLR.  You'll find it sit
 
 In windbg, you'll need mscorwks.dll to load first, and then you can load SOS.  Often, I don't need SOS until well into my debugging session, at which point mscorwks.dll has already been loaded anyway.  However, there are some cases where you'd like SOS loaded at the first possible moment, so you can use some of its commands early (like !bpmd to set a breakpoint on a managed method).  So a surefire way to get SOS loaded ASAP is to have the debugger break when mscorwks gets loaded (e.g., "sxe ld mscorwks").  Once mscorwks is loaded, you can load SOS using the .loadby command:
 
-| 
+|
 ```
 0:000\> **sxe ld mscorwks**
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Debugging - SOS and IDs.md:14]: Same comment as before, about the asterisks and the vertical bar.

File: docs/design/coreclr/profiling/davbr-blog-archive/DoStackSnapshot - HRESULTs.md [hunk 7]
```diff
@@ -5,14 +5,14 @@ Generally, corerror.h tells you all you need to know about what kinds of HRESULT
...
+
+1) No managed frames on stack
+
 If you call DoStackSnapshot when there are no managed functions on your target thread's stack, you can get E\_FAIL.  For example, if you try to walk the stack of a target thread very early on in its execution, there simply might not be any managed frames there yet.  Or, if you try to walk the stack of the finalizer thread while it's waiting to do work, there will certainly be no managed frames on its stack.  It's also possible that walking a stack with no managed frames on it will yield S\_OK instead of E\_FAIL (e.g., if the target thread is jit-compiling the first managed function to be called on that thread).  Again, your code probably doesn't need to worry about all these cases.  If we call your StackSnapshotCallback for a managed frame, you can trust that frame is there.  If we don't call your StackSnapshotCallback, you can assume there are no managed frames on the stack.
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/DoStackSnapshot - HRESULTs.md:12]: This text shows as if it's outside the list item. The rule [MD0029](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md029---ordered-list-item-prefix) is capable of detecting such violations. So, if you want to fix that, I think it worth to be in a separate PR to fix all violations reported by that rule.

File: docs/design/coreclr/profiling/davbr-blog-archive/ELT Hooks - The Basics.md [hunk 8]
```diff
@@ -66,15 +66,15 @@ _(Profiler calls this…)_
...
      _(Profiler implements this…)_
 ```
 typedef UINT_PTR __stdcall FunctionIDMapper(
                 FunctionID funcId,
                 BOOL *pbHookFunction);
 ```
- 
+
 
 2. When FunctionIDMapper is called:
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/ELT Hooks - The Basics.md:79]: This unrelated to the whitespace change, but the first item (Line 8: doesn't appear in GitHub diff, so I can't comment on the exact line) is improperly formatted.

Due to the number of spaces after `1.`, it was treated as a code block, which is completely undesired.

File: docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md [hunk 9]
```diff
@@ -77,50 +77,50 @@ Before I continue from this exciting cliffhanger, a brief interlude.  Everyone k
...
 Now that we’re speaking the same language.  Let’s look at a mixed-mode stack:
 
-| 
+|
 
 Unmanaged
 
  |
-| 
+|
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md:85]: These appear in the same line (both before and after the change), is that desired?
I couldn't guess what is the desired output here.

File: docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md [hunk 10]
```diff
@@ -145,77 +145,77 @@ But before you get too deep, note that the issue of whether and how to seed a st
...
 For the truly adventurous profiler that is doing an asynchronous, cross-thread, seeded stack walk while filling in the unmanaged holes, here’s what it would look like.
 
-| 
+|
 
-Block of  
-Unmanaged  
+Block of
+Unmanaged
 Frames
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md:152]: Are these meant to be in separate lines?

If so, they should be:

```suggestion
Block of\
Unmanaged\
Frames
```

File: docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md [hunk 11]
```diff
@@ -145,77 +145,77 @@ But before you get too deep, note that the issue of whether and how to seed a st
...
+2. You get the target thread’s current register context
+3. You determine if the register context points to unmanaged code (e.g., call ICorProfilerInfo2::GetFunctionFromIP(), and see if you get back a 0 FunctionID)
 4. In this case the register context does point to unmanaged code, so you perform an unmanaged stack walk until you find the top-most managed frame (D)
  |
-| 
+|
 
-Function D  
+Function D
 (Managed)
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md:163]: Same as before, if the intention here is to make these in separate lines, they should be:

```suggestion
Function D\
(Managed)
```

File: docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md [hunk 12]
```diff
@@ -145,77 +145,77 @@ But before you get too deep, note that the issue of whether and how to seed a st
...
 
- | 
+ |
 1. You call DoStackSnapshot with your seed context. CLR suspends target thread again: its suspend count is now 2.  Our sandwich begins.
 
 1. CLR calls your StackSnapshotCallback with FunctionID for D.
  |
-| 
+|
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md:170]: These again appear on the same line before and after the change. I'm not sure about the desired output.

File: docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md [hunk 13]
```diff
@@ -145,77 +145,77 @@ But before you get too deep, note that the issue of whether and how to seed a st
...
 
-Block of  
-Unmanaged  
+Block of
+Unmanaged
 Frames
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md:174]: Same as before.

```suggestion
Block of\
Unmanaged\
Frames
```

File: docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md [hunk 14]
```diff
@@ -3,42 +3,42 @@
...
-MethodRefSig  
-StandAloneMethodSig  
-FieldSig  
-PropertySig  
+MethodDefSig
+MethodRefSig
+StandAloneMethodSig
+FieldSig
+PropertySig
 LocalVarSig
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md:11]: I wouldn't use two space nor `\` for a new line here. This is more suitable to be a list.

```suggestion
- MethodDefSig
- MethodRefSig
- StandAloneMethodSig
- FieldSig
- PropertySig
- LocalVarSig
```

File: docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md [hunk 15]
```diff
@@ -3,42 +3,42 @@
...
 
-Here are the files:  
-[sigparse.cpp](samples/sigparse.cpp) (Rico's signature parser)  
-[sigformat.cpp](samples/sigformat.cpp) (An example extension to the parser)  
+Here are the files:
+[sigparse.cpp](samples/sigparse.cpp) (Rico's signature parser)
+[sigformat.cpp](samples/sigformat.cpp) (An example extension to the parser)
 [PlugInToYourProfiler.cpp](samples/PlugInToYourProfiler.cpp) (Example code to plug the extension into your profiler)
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md:16]: More suitable to be a list rather than explicit new line by either two spaces or a  `\`.

```suggestion
Here are the files:

- [sigparse.cpp](samples/sigparse.cpp) (Rico's signature parser)
- [sigformat.cpp](samples/sigformat.cpp) (An example extension to the parser)
- [PlugInToYourProfiler.cpp](samples/PlugInToYourProfiler.cpp) (Example code to plug the extension into your profiler)
```

File: docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md [hunk 16]
```diff
@@ -3,42 +3,42 @@
...
+            NotifyTypeSimple()
+        NotifyEndType()
+    NotifyEndRetType()
+    NotifyBeginParam()
+        NotifyBeginType()
+            NotifyTypeSimple()
+        NotifyEndType()
+    NotifyEndParam()
+    _… (more parameter notifications occur here if more parameters exist)_
 NotifyEndMethod()
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md:42]: Would you like an explicit new lines by using a `\` or code fencing this whole thing?

File: docs/design/features/additional-deps.md [hunk 17]
```diff
@@ -80,8 +80,8 @@ Where "found" means the version that is being used at run time including roll-fo
 ## 2.1 proposal (add an "any" tfm to store)
 For example,
     `\dotnet\store\x64\any\microsoft.applicationinsights\2.4.0`
-    
-The `any` tfm would be used if the specified tfm (e.g. netcoreapp2.0) is not found:    
+
+The `any` tfm would be used if the specified tfm (e.g. netcoreapp2.0) is not found:
     `\dotnet\store\x64\netcoreapp2.0\microsoft.applicationinsights\2.4.0`
```
Comment (Youssef1313) [inline docs/design/features/additional-deps.md:85]: These was in separate lines, but now are in the same line after removing the trailing whitespaces.
Do you want them to be in separate lines again?

File: docs/design/features/framework-version-resolution.md [hunk 18]
```diff
@@ -30,7 +30,7 @@ In the `.runtimeconfig.json` these values are defined like this:
 ```
 
 #### Framework name
-Each framework reference identifies the framework by its name.  
+Each framework reference identifies the framework by its name.
 Framework names are case sensitive (since they're used as folder names even on Linux systems).
```
Comment (Youssef1313) [inline docs/design/features/framework-version-resolution.md:34]: This was in separate lines, but now will become in the same line. Do you want them to be in separate lines again?

File: docs/design/features/framework-version-resolution.md [hunk 19]
```diff
@@ -147,12 +147,12 @@ Pros
 Cons
 * Testing behavior of new releases with pre-release versions is not fully possible (see below).
 * Some special cases don't work:
-One special case which would not work:  
-*Component A which asks for `2.0.0 LatestMajor` is loaded first on a machine which has `3.0.0` and also `3.1.0-preview` installed. Because it's the first in the process it will resolve the runtime according to the above rules - that is prefer release version - and thus will select `3.0.0`.  
-Later on component B is loaded which asks for `3.1.0-preview LatestMajor` (for example the one in active development). This load will fail since `3.0.0` is not enough to run this component.  
+One special case which would not work:
+*Component A which asks for `2.0.0 LatestMajor` is loaded first on a machine which has `3.0.0` and also `3.1.0-preview` installed. Because it's the first in the process it will resolve the runtime according to the above rules - that is prefer release version - and thus will select `3.0.0`.
```
Comment (Youssef1313) [inline docs/design/features/framework-version-resolution.md:151]: This was in a new line, but after removing the trailing whitespace from the previous line it's now not. Do you want it to be in a separate line?

File: src/mono/mono/mini/cpu-amd64.md [hunk 20]
```diff
@@ -24,9 +24,9 @@
 #
```
Comment (Youssef1313) [inline src/mono/mono/mini/cpu-amd64.md:24]: This file doesn't seem to have any markdown syntax. I think it should be a txt file.

File: eng/docker/Readme.md [hunk 21]
```diff
@@ -5,7 +5,7 @@ Provides reusable docker build infrastructure for the dotnet/runtime repo.
 ## libraries-sdk Dockerfiles
```
Comment (Youssef1313) [inline eng/docker/Readme.md:5]: GitHub doesn't render the diff until I explicitly click "Load diff" for some files, including this one. (It doesn't have a large changes, so that's not the reason)

Could it be some encoding problem?

cc: @sharwell, I think you referred in a comment somewhere that GitHub treats some files as binary if they have incorrect encoding. Can you confirm?

File: docs/workflow/testing/libraries/testing-wasm.md [hunk 22]
```diff
@@ -7,14 +7,14 @@ In order to be able to run tests, the following JavaScript engines should be ins
 
 They can be installed as a part of [jsvu](https://github.com/GoogleChromeLabs/jsvu).
 
-Please make sure that a JavaScript engine binary is available via command line,  
+Please make sure that a JavaScript engine binary is available via command line,
 e.g. for V8:
```
Comment (Youssef1313) [inline docs/workflow/testing/libraries/testing-wasm.md:11]: Do you want these to be in the same line or separate lines?

Before removing the spaces they were separate lines, after removing they are in the same line.

**Comment by Youssef1313** [comment]
@jkotas I've made some manual fixes, and left some questions.

**Comment by CoffeeFlux** [comment]
Thanks for doing this!

File: docs/design/coreclr/profiling/davbr-blog-archive/Debugging - Activation.md (hunk 5 above)
Comment (jkotas) [reply docs/design/coreclr/profiling/davbr-blog-archive/Debugging - Activation.md:21]: The double asterisks are result of some auto-converter. They  should be deleted.

Also, `|` around the block should be deleted.

File: docs/design/coreclr/profiling/davbr-blog-archive/Debugging - SOS and IDs.md (hunk 6 above)
Comment (jkotas) [reply docs/design/coreclr/profiling/davbr-blog-archive/Debugging - SOS and IDs.md:14]: Same as above.

File: docs/design/coreclr/profiling/davbr-blog-archive/Debugging - SOS and IDs.md [hunk 23]
```diff
@@ -35,7 +35,7 @@ As far as your profiler is concerned, a FunctionID is just an opaque number.  It
 
 Ok, so FunctionID = (MethodDesc \*).  How does that help you?  SOS just so happens to have a command to inspect MethodDescs: !dumpmd.  So if you're in a debugger looking at your profiler code that's operating on a FunctionID, it can beneficial to you to find out which function that FunctionID actually refers to.  In the example below, the debugger will break in my proifler's JITCompilationStarted callback and look at the FunctionID.  It's assumed that you've already loaded SOS as per above.
 
-| 
+|
```
Comment (jkotas) [inline docs/design/coreclr/profiling/davbr-blog-archive/Debugging - SOS and IDs.md:38]: I would delete all these `I`.

File: docs/design/coreclr/profiling/davbr-blog-archive/ELT Hooks - The Basics.md (hunk 8 above)
Comment (jkotas) [reply docs/design/coreclr/profiling/davbr-blog-archive/ELT Hooks - The Basics.md:79]: Could you please fix in in this PR?

File: eng/docker/Readme.md (hunk 21 above)
Comment (jkotas) [reply eng/docker/Readme.md:5]: github does this in PRs that modify many files. I think it is the case here.

File: src/mono/mono/mini/cpu-amd64.md (hunk 20 above)
Comment (jkotas) [reply src/mono/mono/mini/cpu-amd64.md:24]: I would exclude them for this PR and deal with these files separately if necessary

File: docs/workflow/testing/libraries/testing-wasm.md (hunk 22 above)
Comment (jkotas) [reply docs/workflow/testing/libraries/testing-wasm.md:11]: This looks fine.

File: docs/design/features/framework-version-resolution.md (hunk 19 above)
Comment (jkotas) [reply docs/design/features/framework-version-resolution.md:151]: Having it all on one line is fine. I would make the raw text to match the rendered view and put it all on the same line in the raw text as well.

Also, could you please change `Some special cases don't work:` -> `Some special cases don't work.` while you are on it.

File: docs/design/features/framework-version-resolution.md (hunk 18 above)
Comment (jkotas) [reply docs/design/features/framework-version-resolution.md:34]: I think it is fine to be on the same line. I would change the raw text to have it on the same line as well.

File: docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md (hunk 16 above)
Comment (jkotas) [reply docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md:42]: code fencing the whole thing

File: docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md (hunk 15 above)
Comment (jkotas) [reply docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md:16]: Agree

File: docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md (hunk 9 above)
Comment (jkotas) [reply docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md:85]: I am not able to find the formatting from the original blog post. I think formatting it like code block would communicate the point in the most readable way:

```
Unmanaged
D (Managed)
Unmanaged
C (Managed)
B (Managed)
Unmanaged
A (Managed)
Main (Managed)
```

File: docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md (hunk 10 above)
Comment (jkotas) [reply docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md:152]: I think it would be fine for the whole "Block of Unmanaged Frames" to be on the single line, in both the rendered view and raw .md.

File: docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md (hunk 11 above)
Comment (jkotas) [reply docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md:163]: Either this, or make it a code block. It would be nice to fix all instances of this pattern the same way.

File: docs/design/features/additional-deps.md (hunk 17 above)
Comment (jkotas) [reply docs/design/features/additional-deps.md:85]: Either way looks fine to me. It would be nice to make the rendered view and raw text look the same.

File: docs/design/coreclr/profiling/davbr-blog-archive/ELT Hooks - tail calls.md [hunk 24]
```diff
@@ -121,9 +121,9 @@ Method 2: On tailcall, "mark" the FunctionID at the top of your stack as needing
 
 With this strategy, for the duration of the call to Three(), the shadow stack will look like this:
 
-Three  
-Helper (marked for deferred pop)  
-Main 
+Three
+Helper (marked for deferred pop)
+Main
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/ELT Hooks - tail calls.md:126]: Explicit new lines or code block?

File: docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md [hunk 25]
```diff
@@ -17,31 +17,31 @@ It’s nice to be able to get call stacks whenever you want them.  But with powe
...
 ```
 And here’s what the CLR calls on your profiler (you can also find this in corprof.idl).  You’ll pass a pointer to your implementation of this function in the callback parameter above.
 ```
-typedef HRESULT \_\_stdcall StackSnapshotCallback(   
-  FunctionID funcId,   
-  UINT_PTR ip,   
-  COR_PRF_FRAME_INFO frameInfo,   
-  ULONG32 contextSize,   
-  BYTE context[],   
+typedef HRESULT \_\_stdcall StackSnapshotCallback(
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Profiler stack walking Basics and beyond.md:30]: @jkotas I believe the backslashes here are extra. Underscores shouldn't be escaped in a codeblock. Can you confirm?

```suggestion
typedef HRESULT __stdcall StackSnapshotCallback(
```

File: docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md [hunk 26]
```diff
@@ -3,43 +3,46 @@
...
+        NotifyBeginType()
+            NotifyTypeSimple()
+        NotifyEndType()
+    NotifyEndRetType()
+    NotifyBeginParam()
+        NotifyBeginType()
+            NotifyTypeSimple()
+        NotifyEndType()
+    NotifyEndParam()
+    _… (more parameter notifications occur here if more parameters exist)_
```
Comment (Youssef1313) [inline docs/design/coreclr/profiling/davbr-blog-archive/Sample A Signature Blob Parser for your Profiler.md:43]: Now that this became a part of code fence, the underscores won't make it italic.

@jkotas Do you want to delete the underscores?

```suggestion
    … (more parameter notifications occur here if more parameters exist)
```

File: docs/workflow/testing/mono/testing.md [hunk 27]
```diff
@@ -51,4 +51,4 @@ You can then, for example, run our HelloWorld sample via:
 dotnet build -c Release $(REPO_ROOT)/src/mono/netcore/sample/HelloWorld
 MONO_ENV_OPTIONS="" COMPlus_DebugWriteToStdErr=1 \
 $(REPO_ROOT)/.dotnet-mono/dotnet $(REPO_ROOT)/src/mono/netcore/sample/HelloWorld/bin/HelloWorld.dll
-```
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply docs/workflow/testing/mono/testing.md:54]: @CoffeeFlux GitHub doesn't show a difference. But it may be a blank line at the end of the file. Not sure.

Here is the code I used to initially generate this PR.

```
void Main()
{	
	var markdown = Directory.GetFiles(@"D:\dotnet\runtime\", "*.md", SearchOption.AllDirectories).ToArray();
	
	for (var i = 0; i < markdown.Length; i++)
	{
	    string[] lines = File.ReadAllLines(markdown[i]);
            File.WriteAllLines(markdown[i], lines.Select(s => s.TrimEnd()));
	}
}
```

**Comment by Youssef1313** [comment]
@jkotas I've done more fixes. Can you review again?

I also just noticed that pipelines run with this PR, although it doesn't touch any code files. The YML configuration for pipelines excludes the following:

```
    - eng/Version.Details.xml
    - .github/*
    - docs/*
    - CODE-OF-CONDUCT.md
    - CONTRIBUTING.md
    - LICENSE.TXT
    - PATENTS.TXT
    - README.md
    - SECURITY.md
    - THIRD-PARTY-NOTICES.TXT
```

shouldn't it exclude ALL markdown files? Not only the mentioned ones? with something like the following:

```
    - eng/Version.Details.xml
    - .github/*
    - docs/*
    - LICENSE.TXT
    - PATENTS.TXT
    - THIRD-PARTY-NOTICES.TXT
    - '/**/*.md'
```

Or are there any code/tests that relies on some md files?

**Comment by Youssef1313** [comment]
@stephentoub @jkotas I also would still recommend disabling trailing_whitespaces from editorconfig for markdown files, and keep it enabled as it's for C# files as there is an analyzer that checks that. Later, if you enabled GH actions and added a markdownlint with MD009 rule enabled, it would be safe to enable trailing_whitespaces for .md again.

**Comment by jkotas** [comment]
> I also just noticed that pipelines run with this PR, although it doesn't touch any code files.
> Or are there any code/tests that relies on some md files?

This can be fixed together with the .md files that are not actually markdown files under mono

**Review by jkotas — APPROVED** [review]
Thank you!
